File: new_modular_javapoet/src/main/java/com/gozap/new_modular_javapoet/MainActivity.java
package com.gozap.new_modular_javapoet;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.gozap.annotation.ARouter;

@ARouter(path="MainActivity")
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: jetpackdemo/src/main/java/com/gozap/jetpack/MainActivity.kt
package com.gozap.jetpack.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.setupWithNavController
import com.gozap.jetpack.R
import kotlinx.android.synthetic.main.activity_main.*

/**
 * https://www.jianshu.com/p/66b93df4b7a6
 */


class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val host = supportFragmentManager.findFragmentById(R.id.my_nav_host_fragment) as NavHostFragment
        val navController = host.navController

        navigation_view.setupWithNavController(navController)
    }


}
